src/identity.js

```diff
@@ -366,10 +382,21 @@ export class Identity extends EventEmitter {
 
         try {
             const autoLoginConverted = autologin ? 1 : 0;
-            let data = await this._hasSession.get('rpc/hasSession.js', { autologin: autoLoginConverted });
-            if (isObject(data.error) && data.error.type === 'LoginException') {
+            let data = null;
+            if (!this._itpMode) {
+                data = await this._hasSession.get('rpc/hasSession.js', { autologin: autoLoginConverted });
+            }
+            if (this._itpMode || (isObject(data.error) && data.error.type === 'LoginException')) {
                 data = await this._spid.get('ajax/hasSession.js', { autologin: autoLoginConverted });
             }
+            if (this._itpModalRequired() && !this._itpMode &&
+                isObject(data.error) && data.error.type === 'UserException' &&
+                this.cache.get(LOGIN_IN_PROGRESS_KEY) !== null) {
```

Comment:
This is a place where I think I would introduce a variable for readability. How about something like:
```javascript
const shouldShowItpModal = this._itpModalRequired() && !this._itpMode &&
    isObject(data.error) && data.error.type === 'UserException' &&
    this.cache.get(LOGIN_IN_PROGRESS_KEY) !== null;
if (shouldShowItpModal) {
    this.cache.delete(...);
    // ...
}
```

Reply:
Fixed!